### PR TITLE
Integrate color picker into PineTimeStyle watchface

### LINKED ISSUE #656: PineTimeStyle allows setting colors that hide UI elements
The PineTime Style watchface step counter (the circle on bottom right of screen) is not working. Can somebody else confirm?

### PR DESCRIPTION
**Pull request by kieranc**
This PR aims to move the color picker functionality into the watchface, to avoid cluttering the settings list and potentially add color settings to other watchfaces.
The settings are now accessed with a long tap on the watchface, which shows a button - if the button is pressed within the timeout (currently 5 seconds), the color picker interface is loaded and can be used to change the colors as before. Otherwise the button disappears, this is to hopefully avoid accidental activation.
I plan to add a bit of logic to change the step gauge needle color depending on the background to fix #656 but I'm having a bit of trouble with my dev watch so I thought I'd create this PR to get some more eyes on the code first.

### CODE REVIEW

**Review by Riksu9000 — COMMENTED** [review]
Looks good. I like this being integrated with the watch face a lot better. The way the buttons appear and disappear probably due to the way the screen is refreshed is interesting.

Seems like going to the settings and swiping to a nonexistent third page crashes the firmware though.

File: src/displayapp/screens/PineTimeStyle.cpp [hunk 1]
```diff
@@ -312,4 +419,135 @@ void PineTimeStyle::Refresh() {
       lv_obj_set_style_local_scale_grad_color(stepGauge, LV_GAUGE_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_WHITE);
     }
   }
+  if (lv_obj_is_visible(btnSet) == true) {
```
[the remark this answers is not in the document]
Comment (kieranc) [reply src/displayapp/screens/PineTimeStyle.cpp:422]: Thanks for that, I had totally missed that lv_obj_get_hidden() existed

File: src/displayapp/screens/PineTimeStyle.cpp [hunk 2]
```diff
@@ -202,6 +300,15 @@ PineTimeStyle::~PineTimeStyle() {
   lv_obj_clean(lv_scr_act());
 }
 
+bool PineTimeStyle::OnTouchEvent(Pinetime::Applications::TouchEvents event) {
+  if (event == Pinetime::Applications::TouchEvents::LongTap) {
+    lv_obj_set_hidden(btnSet, false);
+    savedTick = lv_tick_get();
```
[the remark this answers is not in the document]
Comment (kieranc) [reply src/displayapp/screens/PineTimeStyle.cpp:306]: I'm happy to use whichever you think is more suitable, sounds like probably xTaskGetTickCount()

**Comment by kieranc** [comment]
Thanks for your input Riku, I'm going to sort the gauge needle color later but I have noticed something of a bug if double tap to wake is enabled and I tap the buttons too fast, it turns the display off.. 
I've removed the 3rd screen from Settings.h which should resolve the crash.

**Comment by Riksu9000** [comment]
Doubletapping on the watchface puts the watch to sleep. I believe this can be disabled by returning true in `OnTouchEvent()` when the gesture is doubletap and the buttons are visible.
https://github.com/InfiniTimeOrg/InfiniTime/blob/392c7ad2aceecd8fe976165b66d39f0ed2083911/src/displayapp/DisplayApp.cpp#L230

**Comment by kieranc** [comment]
So much for tidy commits... However, this seems to work as expected, I can't find a better solution to the needle color change so it gets checked when the gauge is created, when the bar BG is changed and on randomize/reset. Not elegant, but it works.

**Review by Riksu9000 — APPROVED** [review]
This includes some changes by me, so someone else should check it too.

**Comment by kieranc** [comment]
So, with help from Riku and others, this finally works as intended. The button closes the picker instead of turning off the screen, and the notification icon and ble icon happily co-exist.

I'm not terribly happy with the code for the icon alignment and am open to suggestions on how to improve it - with the previous [notificationState|bleState].IsUpdated code it seemed to hit the else on the second refresh after displaying the notification icon and realiging the ble icon to the center. By using [notificationState|bleState].Get, it works, the only edge case is that if the ble disconnects and the notification icon is visible, it doesn't move to the center.
I'll give it some more thought tomorrow and see if I can find a better solution but otherwise this is functional and ready for testing.

**Comment by Riksu9000** [comment]
It would've been better to tweak the icons in a separate PR, because this PR was ready to merge, but now we need to work on the new changes too, which are not related to the color picker. If you were to revert those changes and work on them in another branch, we could get this merged quicker.

IsUpdated should be kept, as without it, the icons are refreshed with every cycle even if they haven't changed. It looks like you are aligning the icons next to each other. I think it would be preferrable if all the icons were aligned in a straight line. Have you tried aligning the icons relative to the previous icon, like in the digital watch face? That way when an icon disappears and all the icons are realigned, the later icons move back to fill the empty space.

**Comment by kieranc** [comment]
I could put the notification icon centrally aligned below the BLE icon and it wouldn't be too horrible, but it does need something doing since currently the BLE icon is at 0,25 and the notification icon is 0,40 and they overlap. I'll revert this and make a simpler fix later until i have a proper solution worked out.

**Comment by kieranc** [comment]
Ok it's done, both icon locations are static but they no longer overlap. It'll look a bit odd when there are notifications but no BLE connection, but I'll start working on a better fix. it's already an improvement on overlapping icons, and tbh the notification icon is very rarely displayed unless people accidentally activate silent mode...

image

**Comment by JF002** [comment]
Looks good to me, great work and nice collaboration @kieranc  @Riksu9000  :+1:

**Pull request merged**